pkg/openslosdk/encoding_test.go

```diff
@@ -0,0 +1,252 @@
+package openslosdk
+
+import (
+	"bytes"
+	"embed"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/OpenSLO/OpenSLO/pkg/openslo"
+	v1 "github.com/OpenSLO/OpenSLO/pkg/openslo/v1"
+	"github.com/OpenSLO/OpenSLO/pkg/openslo/v2alpha1"
+)
+
+//go:embed test_data
+var testData embed.FS
+
+func TestDecode(t *testing.T) {
+	t.Run("single map", func(t *testing.T) {
```

Comment:
Should we have a table driven test like the [time package](https://go.dev/src/time/time_test.go) in go. That way we can add cases easily at the top instead of adding a new function everytime

Reply:
yeah, we can switch that to a table test, I'm usually leaning more into this kind of tests setup over table tests when the vast majority of code is not shared between sub tests, in this case the objects' declarations are unique and take up the majority of space, but the assertions are always repeated along with the `Decode` function call, so yeah, I agree :)

Reply:
done :white_check_mark: